build/storage/tests/ut/host-target/fake_pci_sys_fs.py

```diff
@@ -0,0 +1,171 @@
+                    self.get_physfn_path(vf_pci_addr),
+                    self.get_pci_device_path(pf_pci_addr),
+                )
+                self.fs.create_symlink(
+                    self.get_virtfn_path(pf_pci_addr, i),
+                    self.get_pci_device_path(vf_pci_addr),
+                )
+        else:
+            assert False
+
+    def create_pci_device(self, pci_addr):
+        self.fs.create_file(self.get_driver_override_path(pci_addr), contents="(null)")
+        self.fs.create_file(self.get_enable_path(pci_addr), contents="0")
+
+    def create_pci_device_with_sriov(self, pci_addr, max_vfs):
+        self.create_pci_device(pci_addr)
+
+        self.fs.create_file(self.get_sriov_numvfs_path(pci_addr))
+        write_file(self.get_sriov_numvfs_path(pci_addr), 0)
+
+        self.fs.create_file(self.get_sriov_totalvfs_path(pci_addr))
+        write_file(self.get_sriov_totalvfs_path(pci_addr), max_vfs)
+
+        self.fs.create_file(self.get_sriov_drivers_autoprobe_path(pci_addr))
+        write_file(self.get_sriov_drivers_autoprobe_path(pci_addr), 1)
+
+    def create_pci_device_with_bound_driver_and_enabled_sriov(self, pci_addr, max_vfs):
+        self.create_pci_device_with_sriov(pci_addr, max_vfs)
+        self.write_file_with_sysfs_effect(self.get_bind_path(), pci_addr)
+        self.write_file_with_sysfs_effect(
+            self.get_sriov_drivers_autoprobe_path(pci_addr), 0
+        )
+        self.write_file_with_sysfs_effect(
+            self.get_sriov_numvfs_path(pci_addr),
+            max_vfs,
+        )
+
+    def do_not_bind_device_to_driver(self):
+        self.need_to_bind_device_to_driver = False
+
+    def do_not_unbind_device_from_driver(self):
+        self.need_to_unbind_device_from_driver = False
+
+    def bind_pci_device_to_driver(self, pci_addr):
+        self.fs.create_symlink(
+            self.get_bound_device_path(str(pci_addr)),
+            self.get_pci_device_path(pci_addr),
+        )
+        write_file(self.get_enable_path(pci_addr), 1)
+
+    def unbind_pci_device_to_driver(self, pci_addr):
+        write_file(self.get_enable_path(pci_addr), 0)
+        os.remove(self.get_bound_device_path(pci_addr))
+
+    def get_write_file_wrapper(self):
+        def write_file_wrapper(file, content):
+            write_file(file, content)
+
+            if file == self.get_bind_path():
+                if self.need_to_bind_device_to_driver:
+                    self.bind_pci_device_to_driver(PciAddress(str(content)))
+                else:
+                    self.need_to_bind_device_to_driver = True
+            elif file == self.get_unbind_path():
+                if self.need_to_unbind_device_from_driver:
+                    self.unbind_pci_device_to_driver(PciAddress(str(content)))
+                else:
+                    self.need_to_unbind_device_from_driver = True
+            elif os.path.basename(file) == "sriov_numvfs":
+                self._sriov_vfs_set(file, str(content))
```

Comment:
make sure internal function does not get wrong input - validate before num_vfs <= initial_vfs (<= total_vfs)